src/controller.rs

```diff
@@ -114,27 +120,69 @@ async fn reconcile(app_instance: Arc<AppInstance>, ctx: Arc<Context>) -> Result<
         serde_json::from_value(config.metadata.get(PACK_KEY).unwrap().clone())
             .map_err(Error::DecodeKubecfgPackageMetadata)?;
 
-    launch_job(&app_instance, &kubecfg_pack_metadata, Arc::clone(&ctx)).await?;
+    let configmap_name = create_configmap_overlay(&app_instance, Arc::clone(&ctx)).await?;
+    launch_job(
+        &app_instance,
+        &kubecfg_pack_metadata,
+        configmap_name,
+        Arc::clone(&ctx),
+    )
+    .await?;
 
     Ok(Action::await_change())
 }
 
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
+}
+
+async fn create_configmap_overlay(app_instance: &AppInstance, ctx: Arc<Context>) -> Result<String> {
+    let overlay_obj = serde_json::to_string(&app_instance).map_err(Error::RenderOverlay)?;
```

Comment:
I'm still a little fuzzy on whether the serialisation of the `app_instance` is correct for the overlay itself, it seems weird to me that we have the `AppInstance` resource and then use that as the overlay. From what I've pieced together from some internal examples, this does seem to be the case, but perhaps more focussed time spent on this would lead to a different result.

If this is wrong, I think the scaffolding is at least in place for it to be simple to change.

Reply:
yeah we use the entire `AppInstance` as the overlay. This way we don't have to duplicate the reference namespace in the spec and we have access to a bunch of other stuff like image pull secrets etc